clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java

```diff
@@ -58,8 +58,10 @@ public class TopicConfig {
     public static final String FLUSH_MS_CONFIG = "flush.ms";
     public static final String FLUSH_MS_DOC = "This setting allows specifying a time interval at which we will " +
         "force an fsync of data written to the log. For example if this was set to 1000 " +
-        "we would fsync after 1000 ms had passed. In general we recommend you not set " +
-        "this and use replication for durability and allow the operating system's background " +
+        "we would fsync after 1000 ms had passed. Note that this setting depends on the broker-level " +
+        "configuration \"log.flush.scheduler.interval.ms\", which controls how frequently the flush check occurs. " +
+        "If \"log.flush.scheduler.interval.ms\" is not configured, the topic config \"flush.ms\" will not be triggered. " +
```

Comment:
nit: use `<code></code>` instead, so we can have code block for configuration.

Example: 
<img width="625" alt="Screenshot 2025-04-16 at 10 17 55 AM" src="https://github.com/user-attachments/assets/37e7fa4d-a470-4d24-94a9-4af0288070f3" />

Reply:
+1

Reply:
> "If \"log.flush.scheduler.interval.ms\" is not configured, the topic config \"flush.ms\" will not be triggered. "

Maybe we can remove this as the remaining docs is good enough

Reply:
We can also use the `LOG_FLUSH_SCHEDULER_INTERVAL_MS_CONFIG` and `FLUSH_MS_CONFIG` constants instead of hardcoding the strings.

Reply:
@m1a2st The `LOG_FLUSH_SCHEDULER_INTERVAL_MS_CONFIG` is in `org.apache.kafka.server.config.ServerLogConfigs`. It already imported `TopicConfig`. I don't think we can import the `ServerLogConfigs` configuration in `TopicConfig`.

Reply:
> The `LOG_FLUSH_SCHEDULER_INTERVAL_MS_CONFIG` is in `org.apache.kafka.server.config.ServerLogConfigs`

Thanks, but `FLUSH_MS_CONFIG` can update.